Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionControllerDelegatePrivate.h

```diff
@@ -57,11 +57,24 @@ WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA), visionos(WK_XROS_TBA))
  */
 - (void)_webExtensionController:(WKWebExtensionController *)controller receivedTestMessage:(NSString *)message withArgument:(id)argument andSourceURL:(NSString *)sourceURL lineNumber:(unsigned)lineNumber;
 
+/*!
+ @abstract Delegate for the `browser.test.addTest()` JavaScript testing API.
+ @discussion Default implementation logs a message to the system console that a test was added with `testName`. Test harnesses should use this to perform actions when a new test has been added to the queue.
+ */
+- (void)_webExtensionController:(WKWebExtensionController *)controller recordTestAddedWithTestName:(NSString *)testName andSourceURL:(NSString *)sourceURL lineNumber:(unsigned)lineNumber;
+
+/*!
+ @abstract Delegate for the `browser.test.addTest()` JavaScript testing API.
+ @discussion Default implementation logs a message to the system console that a test was started with `testName`. Test harnesses should use this to perform actions at the start of a test.
+ */
+- (void)_webExtensionController:(WKWebExtensionController *)controller recordTestStartedWithTestName:(NSString *)testName andSourceURL:(NSString *)sourceURL lineNumber:(unsigned)lineNumber;
```

Comment:
I think this is still clear and not as redundant.
```suggestion
- (void)_webExtensionController:(WKWebExtensionController *)controller recordTestStartedWithName:(NSString *)testName andSourceURL:(NSString *)sourceURL lineNumber:(unsigned)lineNumber;
```